### PR TITLE
support hbase enable kerberos authentication

### LINKED ISSUE #1227: hbase作为backend是否支持开启kerberos认证？
如题，目前的hbase后端配置里面没有关于kerberos的相关配置。hbase如果开启了kerberos认证，执行init-store.sh初始化hhbase也无法初始化（报错）？是否不支持hbase 开启kerberos？

### PR DESCRIPTION
**Pull request by tmljob**
support hbase enable kerberos authentication.

add configs as follows:
- hbase.kerberos.enable, default value:false
- hbase.krb5.conf,default value:/etc/krb5.conf
- hbase.kerberos.principal,default value:""
- hbase.kerberos.keytab,default value:""

### CODE REVIEW

File: hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java [hunk 1]
```diff
@@ -121,18 +122,44 @@ public synchronized void open() throws IOException {
...
+        boolean isEnableKerberos = config.get(HbaseOptions.HBASE_KERBEROS_ENABLE);
+        Configuration hConfig = null;
+        if(isEnableKerberos) {
+            String krb5Conf = config.get(HbaseOptions.HBASE_KRB5_CONF);
+            System.setProperty("java.security.krb5.conf", krb5Conf);
+            hConfig = HBaseConfiguration.create();
+            String principal = config.get(HbaseOptions.HBASE_KERBEROS_PRINCIPAL);
+            String keyTab = config.get(HbaseOptions.HBASE_KERBEROS_KEYTAB);
+            hConfig.set("hadoop.security.authentication", "kerberos");
+            hConfig.set("hbase.security.authentication","kerberos");
```
Comment (javeme) [inline hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java:134]: keep the origin code and just add line 127~134

Comment (javeme) [inline hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java:134]: missing one space before "kerberos"

File: hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseOptions.java [hunk 2]
```diff
@@ -97,4 +97,36 @@ public static synchronized HbaseOptions instance() {
...
+                    "hbase.kerberos.enable",
+                    "Is Kerberos authentication enabled for HBase.",
+                    disallowEmpty(),
+                    false
+            );
+
+    public static final ConfigOption<String> HBASE_KRB5_CONF =
+            new ConfigOption<>(
+                    "hbase.krb5.conf",
+                    "Krb5.conf configuration file.",
```
Comment (javeme) [inline hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseOptions.java:112]: improve description

File: hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseOptions.java [hunk 3]
```diff
@@ -97,4 +97,36 @@ public static synchronized HbaseOptions instance() {
                     positiveInt(),
                     12 * 60 * 60L
             );
+
+    public static final ConfigOption<Boolean> HBASE_KERBEROS_ENABLE =
+            new ConfigOption<>(
+                    "hbase.kerberos.enable",
```
Comment (javeme) [inline hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseOptions.java:103]: prefer hbase.kerberos_enable, same as other option name

**Review by javeme — CHANGES_REQUESTED** [review]
Thanks for contributing, there are some minor comments: 

1. The PR title is expected to be more clear, like "support hbase enable kerberos authentication".
2. And "implement #1227" to commit message.
3. Some inline comments.

File: hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java [hunk 4]
```diff
@@ -121,18 +122,44 @@ public synchronized void open() throws IOException {
...
-                       config.get(HbaseOptions.HBASE_THREADS_MAX));
+        boolean isEnableKerberos = config.get(HbaseOptions.HBASE_KERBEROS_ENABLE);
+        Configuration hConfig = null;
+        if(isEnableKerberos) {
+            String krb5Conf = config.get(HbaseOptions.HBASE_KRB5_CONF);
+            System.setProperty("java.security.krb5.conf", krb5Conf);
+            hConfig = HBaseConfiguration.create();
+            String principal = config.get(HbaseOptions.HBASE_KERBEROS_PRINCIPAL);
+            String keyTab = config.get(HbaseOptions.HBASE_KERBEROS_KEYTAB);
+            hConfig.set("hadoop.security.authentication", "kerberos");
```
Comment (houzhizhen) [inline hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java:133]: add hConfig.set("hadoop.security.authorization", "true");

File: hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java [hunk 5]
```diff
@@ -121,18 +122,44 @@ public synchronized void open() throws IOException {
...
-                       config.get(HbaseOptions.HBASE_ZK_RETRY));
-
-        // Set hbase.hconnection.threads.max 64 to avoid OOM(default value: 256)
-        hConfig.setInt("hbase.hconnection.threads.max",
-                       config.get(HbaseOptions.HBASE_THREADS_MAX));
+        boolean isEnableKerberos = config.get(HbaseOptions.HBASE_KERBEROS_ENABLE);
+        Configuration hConfig = null;
+        if(isEnableKerberos) {
+            String krb5Conf = config.get(HbaseOptions.HBASE_KRB5_CONF);
+            System.setProperty("java.security.krb5.conf", krb5Conf);
```
Comment (houzhizhen) [inline hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java:129]: remove this line if possible.

Comment (tmljob) [reply hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java:129]: which line need to remobe?pls tell me line num.

File: hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java (hunk 4 above)
Comment (tmljob) [reply hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java:132]: It's not a legal argument. After setting,program reports exception ,as follows:
Exception in thread "main" java.lang.IllegalArgumentException: Invalid attribute value for hadoop.security.authentication of true
	at org.apache.hadoop.security.SecurityUtil.getAuthenticationMethod(SecurityUtil.java:609)
	at org.apache.hadoop.security.UserGroupInformation.initialize(UserGroupInformation.java:272)
	at org.apache.hadoop.security.UserGroupInformation.setConfiguration(UserGroupInformation.java:311)

File: hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java (hunk 5 above)
Comment (javeme) [reply hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseSessions.java:129]: @houzhizhen It seems that setting this environment `java.security.krb5.conf ` is the best way, see [Locating the krb5.conf Configuration File](https://docs.oracle.com/javase/8/docs/technotes/guides/security/jgss/tutorials/KerberosReq.html)

**Review by javeme — APPROVED** [review]

**Review by houzhizhen — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**